servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1478,6 +1518,179 @@ public void setState(long fileId, SetStateOptions options)
     }
   }
 
+  /**
+   * Schedules a file for async persistence.
+   *
+   * @param fileId the id of the file for persistence
+   * @return the id of the worker that persistence is scheduled on
+   * @throws FileDoesNotExistException when the file does not exist
+   */
+  public synchronized long scheduleAsyncPersistence(long fileId) throws FileDoesNotExistException {
```

Comment:
Could we instead lock something more specific than `FileSystemMaster`, e.g. `mWorkerToAsyncPersistFile`?